delfin/drivers/hpe/hpe_msa/ssh_handler.py

```diff
@@ -0,0 +1,465 @@
+                    ser_num = storage_info.get('midplane-serial-number', '')
+                    storage_map = {
+                        'name': storage_info.get('system-name', ''),
+                        'vendor': storage_info.get('vendor-name', ''),
+                        'model': storage_info.get('product-id', ''),
+                        'status': status,
+                        'serial_number': ser_num,
+                        'firmware_version': version_id,
+                        'location': storage_info.get('system-location', ''),
+                        'raw_capacity': int(disk_all),
+                        'total_capacity': int(total_capacity),
+                        'used_capacity': int(volume_all),
+                        'free_capacity': int(total_capacity - volume_all)
+                    }
+            return storage_map
+        except Exception as e:
+            err_msg = "Failed to get storage : %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def list_storage_disks(self, storage_id):
+        try:
+            disk_info = self.ssh_pool.do_exec('show disks')
+            disk_detail = self.handle_detail(disk_info, 'drives')
+            disks_arr = []
+            if disk_detail:
+                for data in disk_detail:
+                    health = data['health']
+                    status = constants.StoragePoolStatus.OFFLINE
+                    if health == 'OK':
+                        status = constants.StoragePoolStatus.NORMAL
+                    size = self.parse_string(data.get('size'))
+                    physical_type = SSHHandler.DISK_PHYSICAL_TYPE.\
+                        get(data.get('description'),
+                            constants.DiskPhysicalType.UNKNOWN)
+                    data_map = {
+                        'native_disk_id': data.get('location', ''),
+                        'name': data.get('location', ''),
+                        'physical_type': physical_type,
+                        'status': status,
+                        'storage_id': storage_id,
+                        'native_disk_group_id': data.get('disk-group', ''),
+                        'serial_number': data.get('serial-number', ''),
+                        'manufacturer': data.get('vendor', ''),
+                        'model': data.get('model', ''),
+                        'speed': data.get('rpm', ''),
+                        'capacity': int(size),
+                        'health_score': status
+                    }
+                    disks_arr.append(data_map)
+            return disks_arr
+        except Exception as e:
+            err_msg = "Failed to get storage disk: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def list_storage_ports(self, storage_id):
+        try:
+            ports_info = self.ssh_pool.do_exec('show ports')
+            ports_arr = ports_info.split('\n')
```

Comment:
What's the difference between `ports_arr` and `ports_array`

Reply:
Variable name changed